src/erlfmt_format.erl

```diff
@@ -318,8 +318,9 @@ binary_op_to_algebra(Op, Meta, Left, Right, Indent) ->
     combine_comments(Meta, maybe_wrap_in_parens(Meta, Doc)).
 
 binary_op_to_algebra(Op, Left, Right, LeftD, RightD, Indent) ->
-    DontBreakCalls = is_call(Right) andalso not has_break_between(Left, Right),
-    case DontBreakCalls orelse is_next_break_fits(Right) of
+    BreakFitsException = is_binary_op_break_fits_exception(Right) andalso
+        not has_break_between(Left, Right),
+    case is_next_break_fits(Right) orelse BreakFitsException of
```

Comment:
I tried it out, so basically adding, `call`, `macro_call` and `bin` to the `NEXT_BREAK_FITS` list, results in:
```erlang
{function, _, [{clause, _, {call, _, _, [Pat]}, empty, [_]}]} = parse_form(
    "f(" ++ String ++ ") -> ok."
)
```
instead of before
```erlang
{function, _, [{clause, _, {call, _, _, [Pat]}, empty, [_]}]} =
    parse_form("f(" ++ String ++ ") -> ok.")
```
and
```
foo(
    bar(
        1
    )
)
```
instead of before
```erlang
foo(bar(
    1
))
```
So I don't think we should modify `is_next_break_fits` and add these cases.

So then there is still a question about whether we should pattern match in `binary_op_to_algebra` to only have this special case for `Op =:= '='`.

Or did I misinterpret what you were trying to say?

Reply:
I was thinking about replicating the conditions that make a tuple be laid out with `line`, 

https://github.com/WhatsApp/erlfmt/blob/15f2a252ed44fb413ade4d006a078666353e25ff/src/erlfmt_format.erl#L395-L396

This means transforming `is_next_break_fits` into something like
```erl
is_next_break_fits({Special, Meta, Values}) when Special =:= tuple; Special =:= bin ->
    has_opening_line_break(Meta, Values) orelse has_trailing_comments(Values);
is_next_break_fits(Other) ->
    %% current implementation
```

This means:
* we retain current behavior for lists, maps, and other `NEXT_BREAK_FITS`
* we still special-case `call` and `macro call` to break after `=` and `::` operators
* for things that are laid out as `flex` we print them like today when they are `flex`, but in cases where they are foced to be multi-line we treat them as any other container

The last point is the only modification

Reply:
This worked very well.  I had no instinct for this yet.  Great idea :)